build.gradle

```diff
@@ -0,0 +1,271 @@
+        //
+        // Finds and reports FQNs used in links in the wrong format.
+        javadocLinkChecker {
+            threshold = 0
+            reportType = "warn"
+        }
+
+        // Right margin line checker.
+        //
+        // Finds and reports about the lines of a code with length above threshold.
+        rightMarginWrappingChecker {
+            maxTextWidth = 100
+        }
+    }
+
+    sourceCompatibility = 1.7
+    targetCompatibility = 1.7
+
+    // Set Java home to point to JDK7 in gradle.properties file.
+    //
+    // For Mac OS X, it looks like this:
+    //
+    // # suppress inspection "UnusedProperty"
+    // org.gradle.java.home=/Library/Java/JavaVirtualMachines/jdk1.7.0_80.jdk/Contents/Home/
+
+    repositories {
+        jcenter()
+        // Google's mirror of Maven Central
+        maven { url = 'https://maven-central.storage.googleapis.com' }
+        mavenCentral()
+        mavenLocal()
+
+        // Spine releases repository.
+        maven { url = spineRepository }
+
+        // Spine snapshots repository.
+        maven { url = spineSnapshotsRepository }
+    }
+
+    dependencies {
+        compile group: 'com.google.guava', name: 'guava', version: guavaVersion
+    }
+
+    sourceSets {
+        main {
+            java.srcDirs = [generatedJavaDir, "$sourcesRootDir/main/java", generatedSpineDir]
+            resources.srcDirs = ["$sourcesRootDir/main/resources", "$generatedRootDir/main/resources"]
+        }
+        test {
+            java.srcDirs = [generatedTestJavaDir, "$sourcesRootDir/test/java", generatedTestSpineDir]
+            resources.srcDirs = ["$sourcesRootDir/test/resources", "$generatedRootDir/test/resources"]
+        }
+    }
+
+    /* Uncomment this block if you need to display console output during the Gradle build.*/
+    test {
+        testLogging {
+            // Make sure output from standard out or error is shown in Gradle output.
+            showStandardStreams = true
+
+            // Or we use events method:
+            // events 'standard_out', 'standard_error'
+
+            // Or set property events:
+            // events = ['standard_out', 'standard_error']
+
+            // Instead of string values we can
+            // use enum values:
+            // events org.gradle.api.tasks.testing.logging.TestLogEvent.STANDARD_OUT,
+            //        org.gradle.api.tasks.testing.logging.TestLogEvent.STANDARD_ERROR,
+        }
+    }
+
+    javadoc {
+        source = sourceSets.main.allJava
+    }
+
+    task sourceJar(type: Jar) {
+        from sourceSets.main.allJava
+        classifier "sources"
+    }
+
+    task testOutputJar(type: Jar) {
+        from sourceSets.test.output
+        classifier "test"
+    }
+
+    task javadocJar(type: Jar, dependsOn: 'javadoc') {
+        from ("$projectDir/build/docs/javadoc")
+        classifier "javadoc"
+    }
+
+    apply from: filterInternalJavadocsScript
+
+    idea.module.iml {
```

Comment:
Can we have these several sections defining `idea.module` merged into one like this?

```groovy
idea {
    module {
        iml {
        }
    }
}
```